src/posthog-core.ts

```diff
@@ -1370,11 +1370,28 @@ export class PostHog {
         this.surveys.renderSurvey(surveyId, selector)
     }
 
-    /** Checks the feature flags associated with this Survey to see if the survey can be rendered. */
+    /**
+     * Checks the feature flags associated with this Survey to see if the survey can be rendered.
+     *
+     * This method is deprecated because it's synchronous and won't return the correct result if the survey is not loaded.
+     * @deprecated Use `canRenderSurveyAsync` instead.
+     * @param surveyId The ID of the survey to check.
+     * @returns A SurveyRenderReason object indicating if the survey can be rendered.
+     */
     canRenderSurvey(surveyId: string): SurveyRenderReason | null {
         return this.surveys.canRenderSurvey(surveyId)
     }
 
+    /**
+     * Checks the feature flags associated with this Survey to see if the survey can be rendered.
+     * @param surveyId The ID of the survey to check.
+     * @param callback A callback function that will be called with the SurveyRenderReason object indicating if the survey can be rendered.
+     * @param forceReload If true, the survey will be reloaded from the server, Default: false
+     */
+    canRenderSurveyAsync(surveyId: string, callback: SurveyRenderReasonCallback, forceReload = false) {
```

Comment:
maybe we can add a default callback to something like:

```suggestion
    canRenderSurveyAsync(surveyId: string, callback: SurveyRenderReasonCallback = (renderReason) => alert(renderReason?.visible
                            ? `Survey can be rendered`
                            : `Survey cannot be rendered: ${renderReason?.disabledReason || 'No reason provided'}`), forceReload = false) {
```

this way we don't always need to pass the callback if we just want to quickly check the value

Reply:
@ioannisj opinions?

Reply:
we don't support IE11 in the default bundle any more so you can use promises
we _should_ autopolyfill and can manually polyfill in the ES5 entrypoint if not

Reply:
aha TIL